engine/time/src/main/java/io/deephaven/time/calendar/BusinessCalendar.java

```diff
@@ -906,14 +1026,30 @@ public int numberBusinessDates(final LocalDate start, final LocalDate end, final
             return NULL_INT;
         }
 
+        if (start.isAfter(end)) {
+            return 0;
+        }
+
+        SummaryData summaryFirst = null;
+        SummaryData summary = null;
         int days = 0;
 
-        for (LocalDate day = start; !day.isAfter(end); day = day.plusDays(1)) {
-            final boolean skip = (!startInclusive && day.equals(start)) || (!endInclusive && day.equals(end));
+        for (Iterator<SummaryData> it = summaryCache.iterator(start, end, startInclusive, endInclusive); it
+                .hasNext();) {
+            summary = it.next();
 
-            if (!skip && isBusinessDay(day)) {
-                days++;
+            if (summaryFirst == null) {
+                summaryFirst = summary;
             }
+
+            days += summary.businessDays;
+        }
+
+        if (summaryFirst == null) {
+            return numberBusinessDatesInternal(start, end, startInclusive, endInclusive);
+        } else {
+            days += numberBusinessDatesInternal(start, summaryFirst.startDate, startInclusive, false);
+            days += numberBusinessDatesInternal(summary.endDate, end, true, endInclusive);
```

Comment:
It's not obvious to me that there isn't some potential over or under counting going on here with inclusive / exclusiveness. 

I _think_ that summaryFirst.startDate and summary.endDate are both *inclusive*, which means I think we want
`days += numberBusinessDatesInternal(summary.endDate, end, false, endInclusive);`?